proposals/TP3-xds-error-propagation.md

```diff
@@ -0,0 +1,121 @@
+
+### Wildcard Subscriptions and Glob Collections
+
+For legacy wildcard subscriptions or [glob collections](TP1-xds-transport-next.md#glob), the control plane is responsible for deciding which resources are included in the subscription. The control plane in this case can provide error details for two different use cases. One when the issue is with the glob itself or later on when the issue is specific to individual resources. 
+
+1. For errors associated with wildcard("" for legacy or `*`) and for xDS-TP glob collections; the control plane `error_details` resource name will match the relevant wildcard request("" or `*` or xDS-TP glob collections). This can be used by the control plane to indicate an error with the collection as a whole.
+2. For errors associated with specific individual resources that match the glob,
+    * The resource name should be the specific resource name associated with the error  OR
+    * The control could just not use this mechanism for wildcard subscriptions, because if the client doesn't have permission to access a resource, then it probably shouldn't be considered to match the wildcard subscription to begin with.
+
+## Rationale
+
+The major alternative to this proposal is to use Wrapped Resources by using Resource Containers defined in https://www.envoyproxy.io/docs/envoy/latest/xds/core/v3/resource.proto#xds-core-v3-resource. 
+
+### Wrapped Resources
+
+xDS resource containers are the default protos used for the Incremental xDS protocol and it's usage in SoTW is controlled via the client feature `xds.config.supports-resource-in-sotw`. 
+
+In this proposal the error information is directly passed as part of the resource field in the Resource Container, using the artifact that the field is a protobuf.Any. This enables us to designate the resource as either a `ResourceError` if the xDS management server encountered problems, or as the actual `Resource` if no errors occurred. 
+
+#### Backward Compatibility
+
+To avoid possible confusion with this behavior, it must be protected with a client feature similar to `supports-resource-in-sotw` called `supports-resource-error-unwrapping`. 
+
+Note: This should also be documented here: https://www.envoyproxy.io/docs/envoy/latest/api/client_features#currently-defined-client-features
+
+#### Why not this approach?
+
+This approach has two major drawbacks compared to the chosen approach:
+
+* This alternative would not work for non-wrapped resources
+* It introduces a backwards compatibility issue, which adding a new field wouldn’t have as clients would just ignore them. 
```

Comment:
Why is that? Isn't this the same as adding the newly proposed field?

Reply:
This is basically saying that it requires the client capability.  Without that, the server cannot send the new fields to the client, because older clients would not know how to interpret a `google.rpc.Status` message inside the `Any` field where the resource should be.

Reply:
I thought it would be in the [resource-wrapper](https://github.com/envoyproxy/envoy/blob/bb35c0c704d63109ebc15c29d96b3ae6f224b290/api/envoy/service/discovery/v3/discovery.proto#L336), and no resource will actually be sent. But I can see why there could be an issue with old-clients that assume that the `resource` (Any) field will be populated by some resource contents.

I think that another drawback of the alternative is for glob-collections, but not sure if it is worth mentioning.

Reply:
I think we'd have the same options for glob collections with this alternative that we do with the proposed approach.